compiler/rustc_target/src/target_features.rs

```diff
@@ -606,9 +614,19 @@ impl super::spec::Target {
     // Returns None if we do not support ABI checks on the given target yet.
     pub fn features_for_correct_vector_abi(&self) -> Option<&'static [(u64, &'static str)]> {
         match &*self.arch {
+            // Tier 1
             "x86" | "x86_64" => Some(X86_FEATURES_FOR_CORRECT_VECTOR_ABI),
             "aarch64" => Some(AARCH64_FEATURES_FOR_CORRECT_VECTOR_ABI),
-            // FIXME: add support for non-tier1 architectures
+            // Tier 2
+            "arm64ec" => Some(AARCH64_FEATURES_FOR_CORRECT_VECTOR_ABI),
+            "arm" => Some(ARM_FEATURES_FOR_CORRECT_VECTOR_ABI),
+            "powerpc" | "powerpc64" => Some(POWERPC_FEATURES_FOR_CORRECT_VECTOR_ABI),
+            "loongarch64" => Some(&[]), // on-stack ABI, so we complain about all by-val vectors
```

Comment:
I don't understand why? "passed by value" is not "not on stack".

Reply:
If everything is passed on the stack, then nothing is passed "by-val" (in the `PassMode::Direct` sense), and therefore the check will never even see a potentially problematic vector on the ABI level. Therefore, if we do see a vector, we can complain.

Reply:
ah, I see.